nmdc_runtime/api/endpoints/nmdcschema.py

```diff
@@ -35,6 +36,41 @@ def get_nmdc_schema_version():
     return version("nmdc_schema")
 
 
+@router.get("/nmdcschema/collection_stats")
+def get_nmdc_database_collection_stats(
+    mdb: MongoDatabase = Depends(get_mongo_db),
+):
+    """
+    Get stats for nmdc:Database MongoDB collections
+
+    Field reference: <https://www.mongodb.com/docs/manual/reference/command/collStats/#std-label-collStats-output>.
+    """
+    present_collection_names = set(nmdc_database_collection_names()) & set(
```

Comment:
It's not clear to me why something more than `mdb.list_collection_names()` is necessary here.

The [docs for that function](https://pymongo.readthedocs.io/en/stable/api/pymongo/database.html#pymongo.database.Database.list_collection_names) say:
> Get a list of all the collection names in this database.

The fact that that set is being combined with another set in this code, suggests to me there may be more collections than those in the database. I don't understand that part (conceptually).

Reply:
One guess: Maybe the combining is being done because `mdb.list_collection_names()` might only include the names of the collections that the Mongo client has access to (permission-wise), which might not be all the ones the end user wants to know the names of. 🤷